### DISCUSSION TITLE
Using of special_login_handler()

### DISCUSSION THREAD

**Discussion by asder30** [ktbyers/netmiko#2620] [Q&A]
With module huawei i try to handle a request for password change.
i do 

> ssh_param = {
>     "username" : username,
>     "host" : ip,
>     "verbose" : True,
>     "password" : password,
>     "session_log" : "./session.log",
>     "auto_connect" : True 
> }
> 
> ssh = HuaweiSSH(**ssh_param)
> ssh.establish_connection()
> ssh.special_login_handler()
> 

But **auto_connect** doesn't work, there was a parallel discussion in 2020. Is there any way to handle such requests?

**Comment by ktbyers** [comment]
@asder30 You are setting "auto_connect" to True, you should be setting it to False.

Here is what typically gets called when "auto_connect" is True:

```python
    def _open(self) -> None:
        """Decouple connection creation from __init__ for mocking."""
        self._modify_connection_params()
        self.establish_connection()
        self._try_session_preparation()
```

**Comment by asder30** [comment]
i've mistaken.
It was an attempt to understand, whether i do everything correct. 
Of course i tried to set **auto_connect** to FALSE. But in this case i got an error

> Traceback (most recent call last):
>   File "huawei_test.py", line 22, in <module>
>     ssh = HuaweiSSH(**ssh_param)
> TypeError: __init__() got an unexpected keyword argument 'auto_connect'
